### PR TITLE
Riot quarterstaff melee weapon for security.

### PR DESCRIPTION
**Pull request by lolman360**
## About The Pull Request

Adds a new weapon choice for Security, the Riot Suppression Electrostaff. It's a huge (cannot into bohs) quarterstaff that needs two hands to hold or use, can be held on back and has two modes. It also has 30 melee block (10 ranged.)

Lethal mode does 20 burn.

Stun mode does 45 stam,1 hardstun and 2 softstun using the knockdown proc (i don't understand bhijin please help) but drains a lot of stam on use.

## Why It's Good For The Game

A new weapon/utility for Sec that's not a straight upgrade (see: AEG, teleshield), but instead an alternative to the stunbaton. As it can take down threats pretty quick it's not a bad choice for things like cult/revs/newcops/gang, but its bulk makes it a logistical challenge normally.

Melee block stops it from getting dominated in CQC by sharpened cult swords or eswords, the small range block offers minor protection against things like op guns/ stolen guns in rev/ gang weapons.

## Changelog
:cl: lolman360, Toriate, Cross
add: NT has approved its prototype Riot Suppression Electrostaff for research and development at your station.
fix: BoHs can no longer hold gigantic items.
soundadd: Sounds for the staff.
imageadd: Sprites for the staff, handcrafted by Toriate.
/:cl:

### CODE REVIEW

File: code/game/objects/items/melee/misc.dm [hunk 1]
```diff
@@ -505,3 +505,82 @@
...
+	icon = 'icons/obj/estaff.dmi'
+	icon_state = "electrostaff_0"
+	item_state = "electrostaff_0"
+	lefthand_file = 'icons/mob/inhands/weapons/estaff_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/weapons/estaff_righthand.dmi'
+	name = "riot suppression electrostaff"
+	desc = "A large quarterstaff, with massive silver electrodes mounted at the end."
+	force = 1
+	damtype = "fire"
+	w_class = WEIGHT_CLASS_HUGE
```
Comment (Trilbyspaceclone) [inline code/game/objects/items/melee/misc.dm]: Fun fact HUGE fits in boh

File: code/game/objects/items/melee/misc.dm [hunk 2]
```diff
@@ -505,3 +505,82 @@
...
+	block_chance = 30
+	light_color = LIGHT_COLOR_YELLOW
+	materials = list(MAT_METAL=1000)
+	hitsound = 'sound/weapons/staff.ogg'
+	attack_verb = list("suppresed", "struck", "beaten", "thwacked", "pulped", "shocked")
+	total_mass = 5
+	var/power = "suppresive"
+	var/lethal = FALSE
+	var/stun_stam_cost_coeff = 1.25
+	var/hardstun_ds = 1
```
Comment (Trilbyspaceclone) [inline code/game/objects/items/melee/misc.dm:537]: Stunbatons do a .8 hardstun or a .08 depending on how maths are done this means you can perma hard stun someone (Like what xenos do)

File: code/game/objects/items/melee/misc.dm [hunk 3]
```diff
@@ -505,3 +505,82 @@
...
+	var/softstun_ds = 2
+	var/stam_dmg = 45
```
Comment (Trilbyspaceclone) [inline code/game/objects/items/melee/misc.dm:539]: Not even a taser bolt does that much... 
Perhaps 30? So its in line with other stun based weaponds - Stun-Gun

File: code/modules/research/techweb/all_nodes.dm [hunk 4]
```diff
@@ -712,6 +712,16 @@
 	research_costs = list(TECHWEB_POINT_TYPE_GENERIC = 2500)
 	export_price = 5000
 
+/datum/techweb_node/electrostaff
+	id = "electrostaff"
+	display_name = "Riot Suppression Equipment"
+	description = "Speak softly and carry a massive stunbaton."
+	prereq_ids = list("adv_weaponry", "electric_weapons", "adv_bluespace")
+	design_ids = list("estaff")
```
Comment (Trilbyspaceclone) [inline code/modules/research/techweb/all_nodes.dm:720]: tbh I might place in Tele-riots in here well placing normal shields in adv weaponds
Or maybe add in basic stun stick as well

File: code/game/objects/items/melee/misc.dm [hunk 5]
```diff
@@ -505,3 +505,82 @@
...
+/obj/item/twohanded/required/electrostaff
+	icon = 'icons/obj/estaff.dmi'
+	icon_state = "electrostaff_0"
+	item_state = "electrostaff_0"
+	lefthand_file = 'icons/mob/inhands/weapons/estaff_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/weapons/estaff_righthand.dmi'
+	name = "riot suppression electrostaff"
+	desc = "A large quarterstaff, with massive silver electrodes mounted at the end."
+	force = 1
+	damtype = "fire"
```
Comment (Trilbyspaceclone) [inline code/game/objects/items/melee/misc.dm]: fire? Why not electric? 
iirc a fire suit may make this weapon useless, or maybe you intent for that?
Then again im not 100% if damtype set to fire looks at melee or fire armor...

Comment (lolman360) [reply code/game/objects/items/melee/misc.dm]: it does burn damage, that's what fire means. All melee damage goes through melee armor,

File: code/game/objects/items/melee/misc.dm (hunk 1 above)
Comment (lolman360) [reply code/game/objects/items/melee/misc.dm]: alright, bumping it up to gigantic

File: code/game/objects/items/melee/misc.dm (hunk 5 above)
Comment (Linzolle) [inline code/game/objects/items/melee/misc.dm]:
```suggestion
	damtype = BURN
```
Use the define.

File: code/game/objects/items/melee/misc.dm (hunk 1 above)
Comment (Linzolle) [reply code/game/objects/items/melee/misc.dm]: I'm pretty sure BoH can hold EVERY item size, just gigantic takes up more room.

**Comment by Linzolle** [comment]
Of note: You mention being able to wear the quarterstaff on "certain armours" however, A. This is not coded in anywhere, and B. Even if it was, suit storage cannot hold anything larger than bulky.

**Comment by lolman360** [comment]
decided against to bring it inline with combat shotgun (its ranged "equivalent")

**Pull request closed without merging**